cpp/src/strings/utilities.cu

```diff
@@ -141,36 +141,74 @@ namespace
 // with RMM initialize/finalize. See PR #3159 for details on this approach.
 __device__ character_flags_table_type character_codepoint_flags[sizeof(g_character_codepoint_flags)];
 __device__ character_cases_table_type character_cases_table[sizeof(g_character_cases_table)];
-std::mutex g_flags_table_mutex, g_cases_table_mutex;
+__device__ special_case_mapping character_special_case_mappings[sizeof(g_special_case_mappings)];
+__device__ character_special_case_hash_indices_table_type character_special_case_hash_indices[sizeof(g_special_case_hash_indices)];
```

Comment:
How much memory overhead is there from all of these lookup tables?

Reply:
Pretty small.   g_special_case_hash_indices is 500 uint_16s, and g_special_case_mappings is 50 special_case_mapping structs.